### PR TITLE
SONARJNKNS-264 Use new 'sonar-scanner' executable in the SQ Scanner step

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: src/main/java/hudson/plugins/sonar/SonarRunnerInstallation.java [hunk 1]
```diff
@@ -70,23 +68,27 @@ public SonarRunnerInstallation(String name, String home, List<? extends ToolProp
...
+        }
         return null;
       }
     });
   }
 
-  private File getExeFile() {
-    String execName = Functions.isWindows() ? "sonar-runner.bat" : "sonar-runner";
+  private File getExeFile(String name) {
+    String execName = Functions.isWindows() ? name + ".bat" : name;
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerInstallation.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by janos-ss** [comment]
I'd add a clarifying parentheses, but it's LGTM

File: src/main/java/hudson/plugins/sonar/SonarRunnerInstallation.java [hunk 2]
```diff
@@ -70,23 +68,27 @@ public SonarRunnerInstallation(String name, String home, List<? extends ToolProp
...
+        }
         return null;
       }
     });
   }
 
-  private File getExeFile() {
-    String execName = Functions.isWindows() ? "sonar-runner.bat" : "sonar-runner";
+  private File getExeFile(String name) {
+    String execName = Functions.isWindows() ? (name + ".bat") : name;
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerInstallation.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS864)

**Pull request merged**